content/2016.12.prototyping-an-app-with-kinto.md

```diff
@@ -0,0 +1,145 @@
+
+In order to build something like that, I'd normally have to fire up a django project, maybe use django-rest-framework to have a proper REST API for the future web or mobile application. Then choose a database like postgresql and configure it.
+
+But I'm not even sure it's a good idea! Maybe we can instead give it a try using the [kinto-admin].
+
+## Let's get started
+
+Let's first think about the data model:
+
+* title: a string
+* comment: a multiline string
+* tag: a string taken from a list of choices
+* price: a float
+
+And then how to store entries in Kinto:
+
+* we'll have a bucket (`track-my-expenses`) which contains a collection (`car-expenses`)
+* a [JSON schema] on the collection will validate the data before it's stored in records
+* and a [UI schema] to improve the way the form is displayed in the admin (eg. a textarea for the comment instead of a single line text input).
+
+Our proof of concept will take advantage of the [kinto-admin] UI which gives us a way to create/read/update/delete entries (records).
+
+
+## Create the bucket and the collection
+
+For our proof of concept, we'll use Mozilla's demo server, which is wiped clean every night: https://kinto.dev.mozaws.net/v1/.
+
+Let's sign in to the [admin] with the user/pass of our choice (the user will be "created" on the spot):
+
+![kinto-admin-sign-in]({filename}/images/kinto-admin-sign-in.png)
+
+Once connected, we can [create a bucket]:
+
+![kinto-admin-connected]({filename}/images/kinto-admin-connected.png)
+![kinto-admin-create-bucket]({filename}/images/kinto-admin-create-bucket.png)
+
+And then [create a collection]:
+
+![kinto-admin-create-collection]({filename}/images/kinto-admin-create-collection.png)
+
+Let's not submit this new collection just yet, we'd first like to add a [JSON schema].
+
+
+## Add a JSON schema
+
+As you can see while creating a collection, we have the option of adding a JSON schema. This schema will be used to validate the data we'll submit, before it's stored in a record.
+It'll also be used to automatically generate the CRUD forms for our data, straight into the admin.
+
+Following our data model, here's what the JSON schema should look like:
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "title": {
+      "type": "string",
+      "title": "Title",
+      "description": "Short title for an expense"
+    },
+    "comment": {
+      "type": "string",
+      "title": "Comment",
+      "description": "Provide details if needed..."
+    },
+    "tag": {
+      "type": "string",
+      "enum": ["fuel", "repair", "insurance", "maintenance", "equipment"],
+      "title": "Tag",
+      "description": "A tag for this expense"
+    },
+    "price": {
+      "type": "number",
+      "title": "Price",
+      "description": "The cost for this expense"
+    }
+  }
+}
+```
+
+Note the `enum` attribute on the tag which will be rendered in the forms as a select, instead of a simple input text field.
```

Comment:
For me tag means that you can select multiple one, in that case I would use category instead of tag.